python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -996,6 +1007,11 @@ def _removeFileExists(uri: ButlerURI) -> None:
                     with formatter._updateLocation(tmpLocation):
                         formatter.write(inMemoryDataset)
                     uri.transfer_from(tmpLocation.uri, transfer="copy", overwrite=True)
+
+                    # Cache if required
+                    if self.cacheManager.should_be_cached(ref):
+                        self.cacheManager.move_to_cache(tmpLocation.uri, ref)
```

Comment:
If this is how we always use these methods, I wonder if they should be combined (not necessarily advocating that; just something to ponder).

Reply:
Good idea. I changed it and that allowed the null cache to return `None` if it didn't want to cache anything.